api/pkg/apis/v1alpha1/providers/target/helm/helm.go

```diff
@@ -315,17 +315,24 @@ func (i *HelmTargetProvider) Get(ctx context.Context, deployment model.Deploymen
 		for _, res := range results {
 			if (deployment.Instance.Spec.Scope == "" || res.Namespace == deployment.Instance.Spec.Scope) && res.Name == component.Component.Name {
 				repo := ""
-				if strings.HasPrefix(res.Chart.Metadata.Tags, "SYM:") { //we use this special metadata tag to remember the chart URL
-					repo = res.Chart.Metadata.Tags[4:]
+				name := ""
+				if strings.HasPrefix(res.Chart.Metadata.Tags, "SYM-REPO:") { //we use this special metadata tag to remember the chart URL
+					parts := strings.Split(res.Chart.Metadata.Tags, ";")
+					if len(parts) != 2 {
+						sLog.ErrorfCtx(ctx, "  P (Helm Target): failed to parse chart metadata tags: %+v", res.Chart.Metadata.Tags)
+						err = v1alpha2.NewCOAError(err, fmt.Sprintf("%s: failed to parse chart metadata tags", providerName), v1alpha2.HelmActionFailed)
+						return nil, err
+					}
+					repo = parts[0][9:]
+					name = parts[1][9:]
 				}
-
 				ret = append(ret, model.ComponentSpec{
 					Name: res.Name,
 					Type: "helm.v3",
 					Properties: map[string]interface{}{
 						"chart": map[string]string{
-							"repo":    repo,
-							"version": res.Chart.Metadata.Version,
+							"repo": repo,
```

Comment:
Seems we don't carry over in version. So, for version upgrade, we will always run helm upgrade, is this right?

Reply:
I think this is the result of changes getting crossed. We use a special tag "SYM:" to remember the chart repo, as I didn't know how to get it back with Helm SDK:

chart.Metadata.Tags = "SYM:" + helmProp.Chart.Repo //this is not used by Helm SDK, we use this to carry repo info

This was later updated to capture chart name as well:

chart.Metadata.Tags = "SYM-REPO:" + helmProp.Chart.Repo + ";SYM-NAME:" + helmProp.Chart.Name

Then, in the Get() method, we are trying to reconstruct repo and name from the tags, as seen in the code. I don't remember why the version is skipped - maybe a merging error on the way.

So, my proposal is:

1) Don't need to capture chart name in the special tag as we can read it back from chart metadata

chart.Metadata.Tags = "SYM:" + helmProp.Chart.Repo

Or, we can skip using the tag if there's a SDK native way to remember the repo

2) Play back the version during Get()

repo := ""
if strings.HasPrefix(res.Chart.Metadata.Tags, "SYM:") { //we use this special metadata tag to remember the chart URL
    repo = res.Chart.Metadata.Tags[4:]

...

"chart": map[string]string{
   "repo":    repo,
   "version": res.Chart.Metadata.Version,
   "name": res.Chart.Metadata.Name,
...

Please also add test cases to Get() after Apply() and verify the information is played back correctly.

Reply:
I checked it. Get will return version name and repo. and other properties
![image](https://github.com/user-attachments/assets/7c80262d-e2d7-417f-94e5-050ba821db3b)